tt_metal/impl/allocator/allocator.hpp

```diff
@@ -127,6 +135,7 @@ struct Allocator {
     std::unordered_map<uint32_t, std::vector<uint32_t>> dram_channel_to_bank_ids;
     std::unordered_map<uint32_t, CoreCoord> bank_id_to_logical_core;
     std::unordered_map<BufferType, std::unordered_map<CoreCoord, std::vector<uint32_t>>> logical_core_to_bank_ids;
+    std::set<Buffer *> allocated_buffers;
```

Comment:
I wonder, why `set`?

Reply:
Since we need to have good erasure performance when deallocating buffers, I chose set (and set keeps it comparable to the original insertion/erasure complexity we had with map). Open to thoughts on alternative container to use

Reply:
do you mean that unordered_set has worse deallocation perofromance?

Reply:
No, the original comment wasn't clear on what the question/issue was, and I thought it was asking why I chose a set-like container, not between using set/unordered_set. I will change to `unordered_set`